plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceData.java

```diff
@@ -0,0 +1,289 @@
+package com.cloud.hypervisor.xenserver.resource;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class CitrixResourceData {
```

Comment:
it is an exercise in reducing the sheer size and complexity of the class. obviously much more needs to be done on this class. I am dropping this for now.